proprietary/design-tokens/build/build-tokens.mjs

```diff
@@ -0,0 +1,244 @@
+ * @returns {Object.<string, Array.<string>>} Object with path as key and a array of combined tokensets
+ */
+const createProducts = (groups) => {
+  const viewports = groups['viewport'].map(({name}) => cleanName(name));
+  const modes = groups['mode'].map(({name}) => cleanName(name));
+  const combined = modes.map((mode) => [mode, [...viewports]]);
+  const mappings = groups['nl-mapping'].map(({name}) => name);
+
+  const logiusProducts = groups['product']
+    .filter(({name}) => cleanName(name) !== 'nldoc')
+    .flatMap(({name, selectedTokenSets}) => {
+      const productCombinations = getProductCombinations(name, combined);
+
+      selectedTokenSets = filterTokenSets(selectedTokenSets);
+      
+      return productCombinations.map(({name, mode, viewport}) => {
+        const modeTokens = getTokensByMode(groups, mode);
+        const viewportTokens = getTokensByViewport(groups, viewport);
+
+        return [`${name}/${mode}/${viewport}`, [...selectedTokenSets, ...modeTokens, ...viewportTokens]];
+      });
+    }
+  );
+
+  const nldocProducts = groups['nldoc']
+    .flatMap(({name, selectedTokenSets}) => {
+      const productCombinations = getProductCombinations(name, combined);
+
+      const nlDocTokenSets = filterTokenSets(groups['product'].find(({name}) => cleanName(name) === 'nldoc').selectedTokenSets);
+      selectedTokenSets = filterTokenSets(selectedTokenSets);
+
+      return productCombinations.map(({name, mode, viewport}) => {
+        const modeTokens = getTokensByMode(groups, mode);
+        const viewportTokens = getTokensByViewport(groups, viewport);
+
+        return [`nldoc/${name ? name + '/' : ''}${mode}/${viewport}`, [...nlDocTokenSets, ...selectedTokenSets, ...modeTokens, ...viewportTokens]];
+      });
+    }
+  );
+
+  return {
+    ...Object.fromEntries(logiusProducts), 
+    ...Object.fromEntries(nldocProducts),
+  };
+}
+
+/**
+ * Our version of permutateThemes 
+ * @param {object} themes parsed JSON object from $themes.json
+ * @returns Object with path as key and a array of combined tokensets
+ */
+const permutateThemes = (themes) => {
+  const themeGroups = getThemeGroupObjects(themes);
+
+  return createProducts(themeGroups);
+};
+
+/**
+ * Builds tokens
+ *
+ * @async
+ * @returns 
+ */
+export const buildTokens = async () => {
+  console.info('Building Tokens');
+
+  const $themes = JSON.parse(await readFile(`${IMPORTED_SRC_FOLDER}/$themes.json`, 'utf-8'));
+  const themes = permutateThemes($themes);
+
+  const configs = Object.entries(themes).map(([name, tokenSets]) => ({
+    source: [
+      // `${MANUAL_SRC_FOLDER}/missingTokens.json`,
+      ...tokenSets.map((tokenSet) => `./**/${tokenSet}.json`),
+    ],
+    include: [
+      './**/core/*.json', 
+      './**/common/**/*.json', 
+      './**/component/*.json', 
+      './**/nl/*.json', // For now, later this will be handled different.
+    ],
+    preprocessors: ['tokens-studio'], // <-- since 0.16.0 this must be explicit
```

Comment:
Is dit al voorsorteren op de volgende update? Momenteel gebruiken we nog `0.15.x`

Reply:
Zo stond het in hun voorbeeld-code, ik dacht, laat het er maar in.